cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -486,15 +486,21 @@ filterConfigureFlags flags cabalLibVersion
   | cabalLibVersion < mkVersion [2,1,0]  = flags_2_1_0
   | cabalLibVersion < mkVersion [2,5,0]  = flags_2_5_0
   | cabalLibVersion < mkVersion [3,7,0]  = flags_3_7_0
+  | cabalLibVersion < mkVersion [3,9,0]  = flags_3_9_0
   | otherwise = error "the impossible just happened" -- see first guard
   where
-    flags_latest = flags        {
+    flags_latest = flags {
       -- Cabal >= 1.19.1 uses '--dependency' and does not need '--constraint'.
       -- Note: this is not in the wrong place. configConstraints gets
       -- repopulated in flags_1_19_1 but it needs to be set to empty for
       -- newer versions first.
       configConstraints = []
-      }
+    }
+
+    flags_3_9_0 = flags {
+        configDependencies = configDependencies flags ++ configPromisedDependencies flags
```

Comment:
I am not in a position to answer this question and will leave it to a future Cabal developer.

Reply:
I thought more about this, and I hink the problem is that reporting the unsupported feature at flag filtering time is too late. I can see three ways to handle this:

1. The good (but more complex) way: have `--enable-multi-repl` introduce a constraint of `Cabal >= 3.11` with appropriate provenance on affected setup components; assert that `--promised-dependency` isn't there in the flag filter.
2. The bad way: just filter out the flag or pass it through, resulting in incomprehensible errors in edge cases.
3. The ugly (but simple) way: don't introduce any constraint, and in the flag filter just call `error` explaining the user how to manually add the constraint.

Reply:
is this the last remaining issue? What would be an error message you would accept

Reply:
I tried (3) and it turned out to be a mess (it was failing in multiple places) so I implemented (1) but on _all_ `setup.Cabal`, not just the closure of repl targets, because it was simpler. I also added a test.